### PR TITLE
feat(PMS-162): Complete FE data realtime reflection of monitoring feature

### PR DESCRIPTION
**Pull request by zKirinz**

### CODE REVIEW

**Review by LeXuanDai0307 — DISMISSED** [review]
Ok

File: src/pages/ProductionManagement/hooks/useBlueprints.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  const [tabValue, setTabValue] = useState(DIAGRAMS[0].value);
+  const tabInfo = useMemo(() => getTabItemByValue(tabValue), [tabValue]);
+  const { blueprints, updateBlueprints, updateRenderedBlueprintId } =
+    useContext(BlueprintsContext);
+  const [isBlueprintReady, setIsBlueprintReady] = useState<boolean>(false);
+
+  const { data: blueprintList } = useQueryBlueprints();
+
+  useEffect(() => {
+    if (blueprintList != undefined) {
```
Comment (oHTGo) [inline src/pages/ProductionManagement/hooks/useBlueprints.ts]:
```suggestion
    if (blueprints) {
```

File: src/pages/ProductionManagement/hooks/useBlueprints.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+} from '~/pages/ProductionManagement/helpers/diagrams';
+
+export default () => {
+  const [tabValue, setTabValue] = useState(DIAGRAMS[0].value);
+  const tabInfo = useMemo(() => getTabItemByValue(tabValue), [tabValue]);
+  const { blueprints, updateBlueprints, updateRenderedBlueprintId } =
+    useContext(BlueprintsContext);
+  const [isBlueprintReady, setIsBlueprintReady] = useState<boolean>(false);
+
+  const { data: blueprintList } = useQueryBlueprints();
```
Comment (oHTGo) [inline src/pages/ProductionManagement/hooks/useBlueprints.ts]:
```suggestion
  const { data: blueprints } = useQueryBlueprints();
```

File: src/pages/ProductionManagement/helpers/diagrams.tsx [hunk 3]
```diff
@@ -130,19 +130,17 @@ export const getTabItemByValue = (tabValue: number): TabItem => {
...
 export function Figures({ figuresCoordinateList, figureValues }: FiguresProps) {
-  return figuresCoordinateList.map(({ displayCoordinate, address }) => {
-    const xCoordinate = displayCoordinate.x;
-    const yCoordinate = displayCoordinate.y;
+  return figuresCoordinateList.map(({ x, y, address }) => {
     return (
       <text
-        key={address + xCoordinate + yCoordinate}
+        key={address + x + y}
         className='main__cls-25'
```
Comment (oHTGo) [inline src/pages/ProductionManagement/helpers/diagrams.tsx]: Remove className

File: src/pages/ProductionManagement/context/BlueprintContext.tsx [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+    </BlueprintsContext.Provider>
+  );
+}
+
+export const useCurrentBlueprintContext = (): IBlueprint => {
+  const { blueprints, renderedBlueprintId } = useContext(BlueprintsContext);
+  const blueprint = blueprints.find(
+    (blueprint) => blueprint.id === renderedBlueprintId,
+  );
+  if (blueprint === undefined)
```
Comment (oHTGo) [inline src/pages/ProductionManagement/context/BlueprintContext.tsx]:
```suggestion
  if (!blueprint)
```

File: src/pages/ProductionManagement/partials/Diagrams/TR11/index.tsx [hunk 5]
```diff
@@ -1,27 +1,27 @@
...
-  const figuresCoordinateList = useFiguresCoordinateContext();
+  const { sensorConfigurations } = useCurrentBlueprintContext();
   const figureValues = useMonitoringStore((state) => state.figureValues);
 
   return (
     <svg
-      data-name='Layer 1'
-      id='Layer_1'
+      data-name='Layer 2'
+      id='Layer_2'
```
Comment (oHTGo) [inline src/pages/ProductionManagement/partials/Diagrams/TR11/index.tsx:15]: Remove all id & data-name attributes from the diagram if it doesn't need

**Review by oHTGo — CHANGES_REQUESTED** [review]

File: src/pages/ProductionManagement/partials/Diagrams/TR11/index.tsx (hunk 5 above)
Comment (zKirinz) [reply src/pages/ProductionManagement/partials/Diagrams/TR11/index.tsx:15]: We need the id for styling, I'll remove the data-name.

**Review by oHTGo — DISMISSED** [review]

**Review by oHTGo — APPROVED** [review]
LGTM, maybe

**Pull request merged**